src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2084,22 +2094,15 @@ bool isRefEscape
                 };
 
                 Debug.Assert(receiver.Type is { });
-                if (receiver is not BoundValuePlaceholderBase && method is not null && receiver.Type?.IsValueType == true)
+                Debug.Assert(receiverIsSubjectToCloning != ThreeState.Unknown);
```

Comment:
>Debug.Assert(receiverIsSubjectToCloning != ThreeState.Unknown); [](http://example.com/codeflow?start=16&length=63)

Would it make sense to add similar asserts in local rewriter when we visit affected nodes? I assume we might not get here for some nodes due to types involved, etc.

Reply:
I've tried this (and fixed synthesized entry point callsite thanks to this). But in general I think it's not that easy - local rewriter sometimes calls itself recursively on already-rewritten nodes, so the assert fails there.

Reply:
I've put the asserts into RefSafetyAnalysis visitor instead. It wouldn't catch the synthesized entrypoint (since ref safety analysis is not called on that), but seems like good enough alternative.